app/database/operator/server_data_operator/handlers/post.ts

```diff
@@ -104,6 +112,127 @@ export const exportedForTest = {
+                const existingScheduledPosts = await queryScheduledPostsForTeam(database, currentTeamId, includeDirectChannelPosts).fetch();
+                const deletedScheduledPostIds = existingScheduledPosts.
+                    filter((post) => !idsFromServer.has(post.id)).
+                    map((post) => post.id);
+
+                if (deletedScheduledPostIds.length > 0) {
+                    scheduledPostsToDelete.push(...await this._deleteScheduledPostByIds(deletedScheduledPostIds, true));
+                }
+
+                const createOrUpdateRawValues = getUniqueRawsBy({raws: scheduledPosts ?? [], key: 'id'}) as ScheduledPost[];
+                scheduledPostsToCreateAndUpdate.push(...await this._createOrUpdateScheduledPost(createOrUpdateRawValues, true));
+                break;
+            }
+        }
+
+        const batch: ScheduledPostModel[] = [...scheduledPostsToDelete, ...scheduledPostsToCreateAndUpdate];
+        if (!prepareRecordsOnly && batch.length) {
+            await this.batchRecords(batch, 'handleScheduledPosts');
+        }
+
+        return batch;
+    };
+
+    _createOrUpdateScheduledPost = async (scheduledPosts: ScheduledPost[], prepareRecordsOnly = false): Promise<ScheduledPostModel[]> => {
+        const processedScheduledPosts = await this.processRecords({
+            createOrUpdateRawValues: scheduledPosts,
+            deleteRawValues: [],
+            tableName: SCHEDULED_POST,
+            fieldName: 'id',
+            shouldUpdate: shouldUpdateScheduledPostRecord,
+        });
+
+        const preparedScheduledPosts = await this.prepareRecords({
+            createRaws: processedScheduledPosts.createRaws,
+            updateRaws: processedScheduledPosts.updateRaws,
+            deleteRaws: processedScheduledPosts.deleteRaws,
+            transformer: transformSchedulePostsRecord,
+            tableName: SCHEDULED_POST,
+        }) as ScheduledPostModel[];
+
+        if (preparedScheduledPosts.length && !prepareRecordsOnly) {
+            await this.batchRecords(preparedScheduledPosts, 'handleScheduledPosts');
+        }
+
+        return preparedScheduledPosts;
+    };
+
+    _deleteScheduledPostByIds = async (scheduledPostIds: string[], prepareRecordsOnly = false): Promise<ScheduledPostModel[]> => {
+        const database: Database = this.database;
+        const scheduledPostsToDelete = await database.get<ScheduledPostModel>(SCHEDULED_POST).query(Q.where('id', Q.oneOf(scheduledPostIds))).fetch();
+        const deleteRawValues: ScheduledPost[] = [];
+        for await (const post of scheduledPostsToDelete) {
+            const schedulePost = await post.toApi(database);
+            deleteRawValues.push(schedulePost);
+        }
+        const processedScheduledPosts = await this.processRecords({
+            createOrUpdateRawValues: [],
+            deleteRawValues,
+            tableName: SCHEDULED_POST,
+            fieldName: 'id',
+        });
```

Comment:
Here you are doing a huge detour for something more simple.

`processRecords` is going to make a query based on the key (`fieldName`, which in this case is `id`) over delete values and do its thing. So your code is doing...
- From a list of ids, get the elements of the table
- To transform them into API objects
- So process records can create a list of ids
- So it can query the table and return those models

So... you can do two things here. Skip the middle man (do not call processRecords and use `scheduledPostsToDelete` on `prepareRecords`), or just build the fake raws out of the ids (`const deleteRawValues = scheduledPostIds.map((v) => {id: v);`).

Reply:
Understood. Updated the code.